### PR TITLE
`Pipeline`: Add lecture chat pipeline connection

### PR DESCRIPTION
**Pull request by sebastianloose**
Add a POST route to connect the lecture chat pipeline, enabling Artemis to send messages directly into the lecture chat system.

## Release Notes

- **New Features**
  - Introduced a new `LectureChatStatusUpdateDTO` class for improved status updates in lecture chat scenarios.
  - Added a `LectureChatCallback` class to enhance handling of status updates during lecture chat processing.
  - Implemented a new endpoint `/lecture-chat/{variant}/run` for executing the lecture chat pipeline.

- **Enhancements**
  - Updated `LectureChatPipeline` to integrate a callback mechanism for better response management.
  - Added `COURSE_LANGUAGE` property to the lecture schema for enhanced data organization.
  - Expanded API capabilities to include lecture chat processing and information about available variants. 

- **Bug Fixes**
  - Streamlined the property handling logic for collection creation in the lecture schema.

### CODE REVIEW

File: app/vector_database/lecture_schema.py [hunk 1]
```diff
@@ -30,7 +30,19 @@ def init_lecture_schema(client: WeaviateClient) -> Collection:
...
+        collection = client.collections.get(LectureSchema.COLLECTION_NAME.value)
+
+        # collection.config.add_property(
+        #    Property(
+        #        name=LectureSchema.COURSE_LANGUAGE.value,
+        #        description="The language of the COURSE",
+        #        data_type=DataType.TEXT,
+        #        index_searchable=False,
+        #    )
+        # )
```
Comment (iyannsch) [inline app/vector_database/lecture_schema.py:42]: Just as code rabbit pointed out, do we still need this? As this condition is only triggered when the `LectureSchema.COLLECTION_NAME` exists in the collection, you can be sure that the returned `collection != null`. Do you want to overwrite it or not?

File: app/web/routers/pipelines.py [hunk 2]
```diff
@@ -121,6 +126,32 @@ def run_text_exercise_chat_pipeline_worker(dto, variant):
         callback.error("Fatal error.", exception=e)
 
 
+def run_lecture_chat_pipeline_worker(dto, variant):
+    try:
+        callback = LectureChatCallback(
+            run_id=dto.settings.authentication_token,
```
Comment (iyannsch) [inline app/web/routers/pipelines.py:132]: Might not be your code, but it seems a bit sketchy that something as confidential as a `authentication token` is used for a less-confidential context such as a random `callback id` 😅 Is it supposed to be like this?

File: app/web/routers/pipelines.py [hunk 3]
```diff
@@ -133,6 +164,16 @@ def run_text_exercise_chat_pipeline(
     thread.start()
 
 
+@router.post(
+    "/lecture-chat/{variant}/run",
+    status_code=status.HTTP_202_ACCEPTED,
+    dependencies=[Depends(TokenValidator())],
+)
+def run_lecture_chat_pipeline(variant: str, dto: LectureChatPipelineExecutionDTO):
+    thread = Thread(target=run_lecture_chat_pipeline_worker, args=(dto, variant))
```
Comment (iyannsch) [inline app/web/routers/pipelines.py:173]: The order of arguments is flipped in this POST endpoint: First `variant`, second `dto`. In the internal methods you use the flipped order. I'd be happy if you could decide and align to one way of doing it ☺️

**Review by iyannsch — CHANGES_REQUESTED** [review]

File: app/web/routers/pipelines.py (hunk 2 above)
Comment (bassner) [reply app/web/routers/pipelines.py:132]: Sounds weird but yes. The token is unique per job

**Review by bassner — DISMISSED** [review]

**Pull request merged**